llvm/lib/Target/AArch64/GISel/AArch64LegalizerInfo.cpp

```diff
@@ -1021,6 +1021,11 @@ AArch64LegalizerInfo::AArch64LegalizerInfo(const AArch64Subtarget &ST)
       .clampNumElements(0, v2s32, v4s32)
       .clampNumElements(0, v2s64, v2s64)
       .moreElementsToNextPow2(0)
+      // Do SMax based lowering for < 128 bits.
+      .customIf([=](const LegalityQuery &Q) {
+        LLT SrcTy = Q.Types[0];
```

Comment:
Should the default implementation of lower make a better guess based on legal operations?

Reply:
The customIf action should also go up below the legal cases